ec2_metadata.py

```diff
@@ -61,6 +61,28 @@ def mac(self):
         return requests.get(METADATA_URL + 'mac').text
 
     @cached_property
+    def network(self):
+        '''The network interface map is keyed by MAC address. This structure will
+           be returned like how it is given in the metadata interface.'''
+        return self.recursive_get_data(METADATA_URL, 'network/interfaces/macs/')
+
+    # helper function, not really designed to be called directly by clients.
```

Comment:
you could just prefix it with an underscore to indicate this, it's a convention in python, however in this case an inner function would even work - yes that works ;)

```
def network...
    def recursive_get...
...
```